erpnext/patches/v15_0/update_invoice_remarks.py

```diff
@@ -8,44 +8,144 @@ def execute():
-		update_remarks("Sales Invoice", doc.name, remarks)
+			Step 1: Update remarks in GL Entries
+			Step 2: Update remarks in Payment Ledger Entries
+			Step 3: Update remarks in Sales Invoice	- Should be last step
+	'''
 
+	###	Step 1:	Update remarks in GL Entries
+	update_sales_invoice_gle_remarks()
+	
+	###	Step 2: Update remarks in Payment Ledger Entries
+	update_sales_invoice_ple_remarks()
+
+	###	Step 3: Update remarks in Sales Invoice
+	update_query = """
+		UPDATE `tabSales Invoice` 
+		SET remarks = concat('Against Customer Order ', po_no) 
+		WHERE po_no <> '' AND docstatus = %(docstatus)s and remarks = %(remarks)s
+	"""
+
+	# Data for update query
+	values = {'remarks': 'No Remarks', 'docstatus': 1}
+
+	# Execute query
+	frappe.db.sql(update_query, values=values, as_dict=0)
+	
 
 def update_purchase_invoice_remarks():
-	pi_list = frappe.db.get_all(
-		"Purchase Invoice",
-		filters={
-			"docstatus": 1,
-			"remarks": "No Remarks",
-			"bill_no": ["!=", ""],
-		},
-		fields=["name", "bill_no"],
-	)
-
-	for doc in pi_list:
-		remarks = _("Against Supplier Invoice {0}").format(doc.bill_no)
-		update_remarks("Purchase Invoice", doc.name, remarks)
-
-
-def update_remarks(doctype, docname, remarks):
-	filters = {
-		"voucher_type": doctype,
-		"remarks": "No Remarks",
-		"voucher_no": docname,
-	}
-
-	frappe.db.set_value(doctype, docname, "remarks", remarks)
-	frappe.db.set_value("GL Entry", filters, "remarks", remarks)
-	frappe.db.set_value("Payment Ledger Entry", filters, "remarks", remarks)
+	'''
+		Update remarks in Purchase Invoice. 
+		Some sites may have very large volume of purchase invoices. 
+		In such cases, updating documents one by one wont be successful, especially during site migration step.
+		Refer to the bug report: https://github.com/frappe/erpnext/issues/43634
+		In this case, a bulk update must be done.
+
+			Step 1: Update remarks in GL Entries
+			Step 2: Update remarks in Payment Ledger Entries
+			Step 3: Update remarks in Purchase Invoice - Should be last step
+	'''
+
+	###	Step 1:	Update remarks in GL Entries
+	update_purchase_invoice_gle_remarks()
+
+	###	Step 2:	Update remarks in Payment Ledger Entries
+	update_purchase_invoice_ple_remarks()
+
+	###	Step 3: Update remarks in Purchase Invoice
+	update_query = """
+		UPDATE `tabPurchase Invoice` 
+		SET remarks = concat('Against Supplier Invoice ', bill_no) 
+		WHERE bill_no <> '' AND docstatus = %(docstatus)s and remarks = %(remarks)s
+	"""
+
+	# Data for update query
+	values = {'remarks': 'No Remarks', 'docstatus': 1}
+
+	# Execute query
+	frappe.db.sql(update_query, values=values, as_dict=0)
+
+
+def update_sales_invoice_gle_remarks():
+
+	##	Update query to update GL Entry - Updates all entries which are for Sales Invoice with No Remarks 
+	update_query = """
+		UPDATE 
+		`tabGL Entry` as gle
+		INNER JOIN `tabSales Invoice` as si 
+		ON gle.voucher_type = 'Sales Invoice' AND gle.voucher_no = si.name AND gle.remarks = %(remarks)s
```

Comment:
Joining GL Entry with Sales Invoice will surely be an issue on large databases.

Would also not recommend using 'remarks' on join condition, as it is a 'Text' field.

Reply:
Any better ideas on how to update this data ? 

The existing script any ways fails.. we have a live example of this.
Moreover, the above queries work - I have executed these queries on my system.. In my system it took ~5 sec to execute the query. 
There are about ~100K invoices in my system. 

On a side note - this data migration should be optional. This is merely updating comments / remarks.

Reply:
@ruthra-kumar @vorasmit - Could you please review and close this PR.

Reply:
@ruthra-kumar @vorasmit - any update on this PR ? 

Let me know if there are any alternate approaches to resolving the upgrade issue.

Reply:
@vv-varun 

Will review and update it by this week